### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by varunagrawal**
This PR fixes a bunch of cmake and compiler warnings.

### CODE REVIEW

File: gtsam/nonlinear/tests/testValues.cpp [hunk 1]
```diff
@@ -175,10 +175,13 @@ TEST(Values, basic_functions)
...
-  values.insert(2, Vector3());
-  values.insert(4, Vector3());
-  values.insert(6, Matrix23());
-  values.insert(8, Matrix23());
+  Matrix23 M1, M2;
+  M1 << 0, 0, 0, 0, 0, 0;
+  M2 << 0, 0, 0, 0, 0, 0;
+  values.insert(2, Vector3(0, 0, 0));
+  values.insert(4, Vector3(0, 0, 0));
+  values.insert(6, M1);
```
Comment (dellaert) [inline gtsam/nonlinear/tests/testValues.cpp:183]: How about Z_2x3 ?

File: gtsam/3rdparty/metis/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8)
```
Comment (dellaert) [inline gtsam/3rdparty/metis/CMakeLists.txt:1]: We can't make changes inside metis. Licensing and upgrade difficulties.

File: gtsam/slam/GeneralSFMFactor.h [hunk 3]
```diff
@@ -70,6 +70,7 @@ class GeneralSFMFactor: public NoiseModelFactor2<CAMERA, LANDMARK> {
 protected:
 
   Point2 measured_; ///< the 2D measurement
+  bool verbose_; ///< Flag for print verbosity
```
Comment (dellaert) [inline gtsam/slam/GeneralSFMFactor.h:73]: Adding a boolean flag for 100k factors in an SFM problem. Problematic. Propose to leave this out of this PR and discuss a better strategy

**Review by dellaert — CHANGES_REQUESTED** [review]
Nice. But. Some issues, in comments.

File: gtsam/3rdparty/metis/CMakeLists.txt (hunk 2 above)
Comment (varunagrawal) [reply gtsam/3rdparty/metis/CMakeLists.txt:1]: Interesting. I saw the git-blame for this CMake file and this file has been updated before, hence I went ahead. Does this new information change anything?

File: gtsam/nonlinear/tests/testValues.cpp (hunk 1 above)
Comment (varunagrawal) [reply gtsam/nonlinear/tests/testValues.cpp:183]: `Z_2x3` isn't defined but I see what you mean.

File: gtsam/3rdparty/metis/CMakeLists.txt (hunk 2 above)
Comment (dellaert) [reply gtsam/3rdparty/metis/CMakeLists.txt:1]: Actually, maybe the Makefile is what we added, and we should not make changes in the library itself.

File: gtsam/slam/GeneralSFMFactor.h (hunk 3 above)
Comment (varunagrawal) [reply gtsam/slam/GeneralSFMFactor.h:73]: Alright. Should I leave the exception printing in? Ideally that should be part of a logging framework. I'll already looked up some options.

Comment (dellaert) [reply gtsam/slam/GeneralSFMFactor.h:73]: No, no printing. Let's leave for when we have logging, which would be great. I do like glog :-)

File: gtsam/3rdparty/metis/CMakeLists.txt (hunk 2 above)
Comment (varunagrawal) [reply gtsam/3rdparty/metis/CMakeLists.txt:1]: Yup, the library is unchanged. :slightly_smiling_face: Guess we'll keep this update.

File: gtsam/slam/GeneralSFMFactor.h (hunk 3 above)
Comment (varunagrawal) [reply gtsam/slam/GeneralSFMFactor.h]: Yeah glog and CodeCov (for unit test code coverage) are two things on my radar. I'll create an issue for it that someone else can tackle.

**Review by dellaert — APPROVED** [review]

**Pull request merged**